### DISCUSSION TITLE
Consider Vega or Vega-Lite

### DISCUSSION THREAD

**Discussion by colinmegill** [FormidableLabs/victory#2102] [General]
Victory might consider a mapping to Vega or Vega-Lite schemas. There is an existing react project that consumes and renders the Vega spec: http://vega.github.io/react-vega/?path=/story/react-vega--vegalite-with-layers

Possible benefits of consuming Vega include an opportunity to gain mindshare from the data science community and those moving between Jupyter environments and the web. Currently, that niche is occupied by projects that have emerged from the python community, not all of which are optimized for modern web applications.

https://vega.github.io/vega-lite/
https://vega.github.io/vega/

**Comment by boygirl** [comment]
Hi Colin! It's been an age! This is a neat idea. I'd love to make Victory more appealing to the data science community. I looked at the vega spec a little bit, and I think a lot of it lends itself to a straightforward mapping into Victory components. I struggle to know what to do with vega's concept of signals, though. It also looks like Vega handles data loading, which would be a pretty significant departure for Victory.

**Comment by colinmegill** [comment]
@boygirl your work on Victory is simply incredible :D 

Yes... My feeling was this would only be possible, ultimately, with a subset of Vega / Vega Lite functionality. Assuming the user is doing relatively simple things / showing dots / showing bars, it might be possible to offer some pipeline for interop between Jupyter and web. I think it's possible, but I might be optimistic :) that users who are moving back and forth wouldn't be looking to implement some of the more complex features in Jupyter — ie., if you have web at all, it's to do those things. 

Perhaps it's safe to assume that Jupyter users who want lots of interactivity on that platform aren't the ones porting to full web experiences? Unsure. There are some sites that I've seen that try to capture relative usage — https://python.libhunt.com/bokeh-alternatives matplotlib, altair and seaborn, along with bokeh and, of course, plotly, have mindshare, I'm unsure how vega stacks up.

An aside, I had a _minor_ upgrade on an old project and everything still worked. +1 for API consistency.

![image](https://user-images.githubusercontent.com/1770265/85215974-4a61d280-b34d-11ea-9ab3-3305358e9199.png)
